### PR TITLE
Change catching ValueError and AttributeError to DatabaseCustomError

### PR DESCRIPTION
**Pull request by sunank200**
# Description
## What is the current behavior?

Currently in Snowflake load_file, ValueError and AttributeError are being catched at function level but it should be caught at the line and thrown that error.

closes: #583 

## What is the new behavior?

- Create class `DatabaseCustomError` to raise generic errors like ValueError and AttributeError 
- Raise the specific errors and change the tests.

## Does this introduce a breaking change?
No

### Checklist
- [x] Created tests which fail without the change (if possible)
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

File: src/astro/databases/google/bigquery.py [hunk 1]
```diff
@@ -335,7 +333,9 @@ def get_project_id(self, target_table) -> str:
         try:
             return str(self.hook.project_id)
         except AttributeError:
```
Comment (utkarsharma2) [inline src/astro/databases/google/bigquery.py:335]:
```suggestion
        except AttributeError as e:
```

Comment (utkarsharma2) [inline src/astro/databases/google/bigquery.py:335]:
```suggestion
        except AttributeError as e:
```

File: src/astro/databases/google/bigquery.py [hunk 2]
```diff
@@ -335,7 +333,9 @@ def get_project_id(self, target_table) -> str:
...
-            raise ValueError(f"conn_id {target_table.conn_id} has no project id")
+            raise DatabaseCustomError(
+                f"conn_id {target_table.conn_id} has no project id"
+            )
```
Comment (utkarsharma2) [inline src/astro/databases/google/bigquery.py:338]:
```suggestion
            raise DatabaseCustomError(
                f"conn_id {target_table.conn_id} has no project id"
            ) from e
```

File: src/astro/databases/snowflake.py [hunk 3]
```diff
@@ -419,7 +418,11 @@ def load_file_to_table_natively(
         sql_statement = (
             f"COPY INTO {table_name} FROM @{stage.qualified_name}/{file_path}"
         )
-        self.hook.run(sql_statement)
+        try:
+            self.hook.run(sql_statement)
+        except (ValueError, AttributeError) as exe:
+            logging.warning(exe)
+            raise DatabaseCustomError
```
Comment (utkarsharma2) [inline src/astro/databases/snowflake.py:425]:
```suggestion
            raise DatabaseCustomError from exe
```

File: src/astro/databases/base.py [hunk 4]
```diff
@@ -24,6 +24,15 @@
 from astro.sql.table import Metadata, Table
 
 
+class DatabaseCustomError(ValueError, AttributeError):
```
Comment (utkarsharma2) [inline src/astro/databases/base.py:27]: @sunank200 Any reason for inheriting from ValueError and AttributeError as opposed to Exception?

Comment (sunank200) [reply src/astro/databases/base.py:27]: These are the relevant errors which we want to catch in the location.

File: src/astro/databases/snowflake.py (hunk 3 above)
Comment (sunank200) [reply src/astro/databases/snowflake.py]: [Done](https://github.com/astronomer/astro-sdk/pull/595/commits/5d17ca3c1154b68cbfa3d678733a265efbdcc3f2)

File: src/astro/databases/google/bigquery.py (hunk 2 above)
Comment (sunank200) [reply src/astro/databases/google/bigquery.py]: [Done](https://github.com/astronomer/astro-sdk/pull/595/commits/5d17ca3c1154b68cbfa3d678733a265efbdcc3f2)

File: src/astro/databases/google/bigquery.py (hunk 1 above)
Comment (sunank200) [reply src/astro/databases/google/bigquery.py]: [Done](https://github.com/astronomer/astro-sdk/pull/595/commits/5d17ca3c1154b68cbfa3d678733a265efbdcc3f2)

**Review by utkarsharma2 — APPROVED** [review]

**Pull request merged**